tests/unit/python/foglamp/common/test_configuration_manager.py

```diff
@@ -0,0 +1,430 @@
+        c = ConfigurationManager()
+        assert hasattr(c, '_storage')
+        assert c._storage is storageClientMock
+        assert hasattr(c, '_registered_interests')
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_register_interest_no_category_name(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        with pytest.raises(ValueError) as excinfo:
+            c.register_interest(None,'callback')
+        assert 'Failed to register interest. category_name cannot be None' in str(excinfo.value)
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_register_interest_no_callback(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        with pytest.raises(ValueError) as excinfo:
+            c.register_interest('name',None)
+        assert 'Failed to register interest. callback cannot be None' in str(excinfo.value)
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_register_interest(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        c.register_interest('name','callback')
+        assert 'callback' in c._registered_interests['name']
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_unregister_interest_no_category_name(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        with pytest.raises(ValueError) as excinfo:
+            c.unregister_interest(None,'callback')
+        assert 'Failed to unregister interest. category_name cannot be None' in str(excinfo.value)
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_unregister_interest_no_callback(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        with pytest.raises(ValueError) as excinfo:
+            c.unregister_interest('name',None)
+        assert 'Failed to unregister interest. callback cannot be None' in str(excinfo.value)
+        ConfigurationManagerSingleton._shared_state = {}
+
+    def test_unregister_interest(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        c.register_interest('name','callback')
+        c.unregister_interest('name','callback')
+        assert len(c._registered_interests) is 0
+        ConfigurationManagerSingleton._shared_state = {}
+
+    @pytest.mark.asyncio
+    async def test_run_callback(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        c.register_interest('name','configuration_manager_callback')
+        await c._run_callbacks('name')
+        ConfigurationManagerSingleton._shared_state = {}
+       
+
+    """ Configuration Manager
```

Comment:
can we please remove this stuff? it should be in module doscstring of actual file.

if this block is separate, should we a new Class TestC...  and annotate with allure